src/app/CommandHandler.cpp

```diff
@@ -86,20 +101,20 @@ void CommandHandler::OnInvokeCommandRequest(Messaging::ExchangeContext * ec, con
 
     // NOTE: we already know this is an InvokeCommand Request message because we explicitly registered with the
     // Exchange Manager for unsolicited InvokeCommand Requests.
-    mExchangeCtx.Grab(ec);
+    mResponseSender.SetExchangeContext(ec);
```

Comment:
Suggest that `SetExchangeContext` call `ec.WillSendMessage()` internally, rather than making it explicit as on line 113 below, as there is no case where we wouldn't need to call it from here on down.

Reply:
Discussed offline that this can be done in a followup PR. Added TODO comment and added to [GH issue tracker](https://github.com/project-chip/connectedhomeip/issues/30453)